src/main/java/com/geronimo/service/UserDetailsService.java

```diff
@@ -0,0 +1,81 @@
+package com.geronimo.service;
+
+import com.geronimo.model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+public class UserDetailsService implements org.springframework.security.core.userdetails.UserDetailsService {
```

Comment:
This class is used by spring security to find a user when you try to login by username, and you have to give it authorities that user possesses (getAuthorities method)